src/messaging/Channel.h

```diff
@@ -0,0 +1,241 @@
+    ChannelBuilder & SetPeerKeyID(uint16_t keyId)
+    {
+        assert(mSessionType == SessionType::kCASE);
+        mSessionParameters.mCaseParameters.mPeerKeyId = keyId;
+        return *this;
+    }
+
+    uint32_t GetPeerSetUpPINCode() const { return mSessionParameters.mPaseParameters.mPeerSetUpPINCode; }
+    ChannelBuilder & SetPeerSetUpPINCode(uint32_t peerPINCode)
+    {
+        assert(mSessionType == SessionType::kPASE);
+        mSessionParameters.mPaseParameters.mPeerSetUpPINCode = peerPINCode;
+        return *this;
+    }
+
+    Optional<Inet::IPAddress> GetForcePeerAddress() const { return mForcePeerAddr; }
+    ChannelBuilder & SetForcePeerAddress(Inet::IPAddress peerAddr)
+    {
+        mForcePeerAddr.SetValue(peerAddr);
+        return *this;
+    }
+
+private:
+    NodeId mPeerNodeId                       = kUndefinedNodeId;
+    TransportPreference mTransportPreference = TransportPreference::kDefault;
+    SessionType mSessionType                 = SessionType::kDefault;
+    union SessionParameters
+    {
+        struct
+        {
+            uint16_t mPeerKeyId;
+        } mCaseParameters;
+        struct
+        {
+            uint32_t mPeerSetUpPINCode;
+        } mPaseParameters;
+    } mSessionParameters;
+
+    Optional<Inet::IPAddress> mForcePeerAddr;
+};
+
+class ExchangeContext;
+class ExchangeDelegate;
+
+enum class ChannelState
+{
+    kNone,
+    kPreparing,
+    kReady,
+    kClosed,
+    kFailed,
+};
+
+class ChannelContextHandleAssociation;
+
+/**
+ *  @brief
+ *    ChannelHandle is a reference to a channel. An active ChannelHandle will
+ *    keep the channel available and ready for use, such that a message can be
+ *    sent immediately to the peer.
+ *
+ *    The ChannelHandle controls the lifespan of the channel. When the handle
+ *    is released, the channel will be flagged as pending close, and if there
+ *    is no active exchange which is using the channel, the channel will be
+ *    closed.
+ *
+ *    The ChannelHandle will track channel status, and notify applications
+ *    when the channel state changes via ChannelDelegate.
+ */
+class ChannelHandle
+{
+public:
+    explicit ChannelHandle(ChannelContextHandleAssociation * association = nullptr) : mAssociation(association) {}
+    ~ChannelHandle() { Release(); }
+
+    // non copyable
+    ChannelHandle(const ChannelHandle &) = delete;
+    ChannelHandle & operator=(const ChannelHandle &) = delete;
+
+    // movable
+    ChannelHandle(ChannelHandle && that)
+    {
+        Release();
+        this->mAssociation = that.mAssociation;
+        that.mAssociation  = nullptr;
+    }
+    ChannelHandle & operator=(ChannelHandle && that)
+    {
+        Release();
+        this->mAssociation = that.mAssociation;
+        that.mAssociation  = nullptr;
+        return *this;
+    }
+
+    /*
+     * @brief
+     *   Get the state of the channel (@see ChannelContext).
+     */
```

Comment:
Please remove comments that do not add additional information:

GetState obviously 'Get the state of the channel' and `Release` obviously releases the channel.